src/main/java/org/osiam/storage/query/UserSimpleFilterChain.java

```diff
@@ -44,53 +43,42 @@
     private ExtensionQueryField extensionFilterField;
 
     public UserSimpleFilterChain(CriteriaBuilder criteriaBuilder, ExtensionDao extensionDao,
-                                 ScimExpression scimExpression) {
+                                 FilterExpression filterExpression) {
         this.criteriaBuilder = criteriaBuilder;
         this.extensionDao = extensionDao;
-        this.scimExpression = scimExpression;
+        this.filterExpression = filterExpression;
 
-        String field = scimExpression.getField();
 
-        userFilterField = UserQueryField.fromString(field.toLowerCase(Locale.ENGLISH));
-
-        // It's not a known user field, so try to build a extension filter
-        if (userFilterField == null) {
-            extensionFilterField = getExtensionFilterField(field.toLowerCase(Locale.ENGLISH));
-            if (extensionFilterField == null) {
-                throw new IllegalArgumentException(String.format("Filtering not possible: '%s' not available", field));
```

Comment:
You moved this error handling into `createPredicateAndJoin`. I think failing fast here is a more viable approach. You have to consider, that `createPredicateAndJoin` gets called very late, at a point when all `FilterChain`s and `FilterExpression`s have already been created. So multiple database accesses, to get the extensions fields, might already have happened. But all this would be thrown away, if there is a wrong field in the filter expression.